src/modules/socketClient/SocketClient.ts

```diff
@@ -0,0 +1,42 @@
+import { io, Socket } from "socket.io-client";
+
+class SocketClient {
+
+  socket: Socket
+  isSocketConnected: boolean
+  
+  constructor() {
+    this.isSocketConnected = false;
+    this.socket = io("http://localhost:5000");
+    
+    this.checkSocketConnection().then(() => {
+      this.isSocketConnected = this.socket.connected
+    })
+  }
+
+  async checkSocketConnection(): Promise<any>{
+    await new Promise((resolve, reject) => {
+      this.socket.on('connect', () => {
+        console.log('Socket Connected? ', this.socket.connected);
+        resolve(this.socket.connected)
```

Comment:
Why are we returning `socket.connected` ?

Reply:
This is required to check if the connection is successful. This value is required to avoid multiple socket connections on re-render. 
A globally(to specific routes) available socket can resolve this but still, we would need to check this value somewhere to avoid multiple socket connections.

Reply:
Can you check if this is due to re-connection or multiple sockets being created?